tests/contrib/pynamodb/test_pynamodb.py

```diff
@@ -0,0 +1,175 @@
+from ddtrace.contrib.pynamodb.patch import patch, unpatch
+
+# testing
+from ... import TracerTestCase, assert_is_measured
+from moto.dynamodb import dynamodb_backend
+
+
+class PynamodbTest(TracerTestCase):
+
+    TEST_SERVICE = "pynamodb"
+
+    def setUp(self):
+        patch()
+
+        self.conn = Connection(region="us-east-1")
+        self.conn.session.set_credentials("aws-access-key", "aws-secret-access-key", "session-token")
+
+        super(PynamodbTest, self).setUp()
+        Pin.override(self.conn, tracer=self.tracer)
+
+    def tearDown(self):
+        super(PynamodbTest, self).tearDown()
+        unpatch()
+
+    @mock_dynamodb
+    def test_list_tables(self):
+        dynamodb_backend.create_table("Test", hash_key_attr="content", hash_key_type="S")
+        self.conn.list_tables()
+
+        spans = self.get_spans()
+
+        assert spans
+        span = spans[0]
+
+        assert span.name == "pynamodb.command"
+        assert span.service == "pynamodb"
+        assert span.resource == "ListTables"
+        assert len(spans) == 1
+        assert_is_measured(span)
+        assert span.span_type == "http"
+        assert span.get_tag("aws.operation") == "ListTables"
+        assert span.get_tag("aws.region") == "us-east-1"
+        assert span.get_tag("aws.agent") == "pynamodb"
+        assert span.duration >= 0
+        assert span.error == 0
+
+    @mock_dynamodb
+    def test_delete_table(self):
+        dynamodb_backend.create_table("Test", hash_key_attr="content", hash_key_type="S")
+        self.conn.delete_table("Test")
+        spans = self.get_spans()
+
+        assert spans
+        span = spans[0]
+
+        assert span.name == "pynamodb.command"
+        assert span.service == "pynamodb"
+        assert span.resource == "DeleteTable Test"
+        assert len(spans) == 1
+        assert_is_measured(span)
+        assert span.span_type == "http"
+        assert span.get_tag("aws.operation") == "DeleteTable"
+        assert span.get_tag("aws.region") == "us-east-1"
+        assert span.get_tag("aws.agent") == "pynamodb"
+        assert span.duration >= 0
+        assert span.error == 0
+
+    @mock_dynamodb
+    def test_scan(self):
+        dynamodb_backend.create_table("Test", hash_key_attr="content", hash_key_type="S")
+        self.conn.scan("Test")
+
+        spans = self.get_spans()
+
+        assert spans
+        span = spans[0]
+
+        assert span.name == "pynamodb.command"
+        assert span.service == "pynamodb"
+        assert span.resource == "Scan Test"
+        assert len(spans) == 1
+        assert_is_measured(span)
+        assert span.span_type == "http"
+        assert span.get_tag("aws.operation") == "Scan"
+        assert span.get_tag("aws.region") == "us-east-1"
+        assert span.get_tag("aws.agent") == "pynamodb"
+        assert span.duration >= 0
+        assert span.error == 0
+
+    @mock_dynamodb
+    def test_scan_on_error(self):
+        try:
+            self.conn.scan("OtherTable")
```

Comment:
Instead of using a try..except here we should use

```python
with pytest.raises(pynamodb.exceptions.ScanError):
    self.conn.scan("OtherTable")

spans = self.get_spans()
assert spans
span = spans[0]
assert span.name == "pynamodb.command"
# rest of the assertions
```

This ensures that the exception is raised and will fail the test if it's not. This is important because if the exception is actually not raised then the test will pass without testing anything 🙂

Reply:
Thanks! The latest commit should have this update!